.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,70 @@
+# For most projects, this workflow file will not need changing; you simply need
+# to commit it to your repository.
+#
+# You may wish to alter this file to override the set of languages analyzed,
+# or to provide custom queries or build logic.
+#
+# ******** NOTE ********
+# We have attempted to detect the languages in your repository. Please check
+# the `language` matrix defined below to confirm you have the correct set of
+# supported CodeQL languages.
+#
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ master ]
+  schedule:
+    - cron: '32 22 * * 3'
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'java' ]
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
+      with:
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
+
+    # ℹ️ Command-line programs to run using the OS shell.
+    # 📚 https://git.io/JvXDl
+
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
```

Comment:
```suggestion
    - name: Build
      run: |
        git submodule update --init
        ./remap.sh
        mvn clean package --file $GITHUB_WORKSPACE/pom.xml
```

Reply:
the analysis isn’t failing currently so i see this as unnecessary (not to be confused with the check, which is giving valid security alerts)

Reply:
This should fail though, since without it, the minecraft-server dependency will never be found. So I still think it's better to build manually anyway, just in case.

Reply:
> This should fail though, since without it, the minecraft-server dependency will never be found. So I still think it's better to build manually anyway, just in case.

Elmaker's certificate was fixed but just in case it expires again this should be kept.

Reply:
As far as I'm aware that expiry was related to LetsEncrypt’s recent CA expiry and is extremely unlikely to reoccur (correct me if I’m wrong though!)

Reply:
> As far as I'm aware that expiry was related to LetsEncrypt’s recent CA expiry and is extremely unlikely to reoccur (correct me if I’m wrong though!)

Definitely not, the CA expiry was in September while Elmaker's certificate expired in November.

Reply:
In my opinion it’s unlikely to happen again as they probably added a cronjob for it, however if it expires again I’ll make the change (or a maintainer can do this either on my branch now, or in future on the master branch). Thanks though!